#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_newplugin.dpatch by Thomas Schmidt <tschmidt@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch for the newplugin-script, to not require a normal 
## DP: vdr-source-dir and to add -fPIC to the CXXFLAGS of a
## DP: new plugin

@DPATCH@
diff -urNad vdr-1.3.23/newplugin /tmp/dpep.xLpfdk/vdr-1.3.23/newplugin
--- vdr-1.3.23/newplugin	2005-01-30 14:50:05.000000000 +0100
+++ /tmp/dpep.xLpfdk/vdr-1.3.23/newplugin	2005-04-09 21:28:38.310427464 +0200
@@ -24,7 +24,7 @@
 $PLUGIN_DESCRIPTION = "Enter description for '$PLUGIN_NAME' plugin";
 $PLUGIN_MAINENTRY = $PLUGIN_CLASS;
 
-$PLUGINS_SRC = "PLUGINS/src";
+$PLUGINS_SRC = "./";
 
 $README = qq
 {This is a "plugin" for the Video Disk Recorder (VDR).
@@ -71,7 +71,7 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual
+CXXFLAGS ?= -fPIC -O2 -Wall -Woverloaded-virtual
 
 ### The directory environment:
 
@@ -239,7 +239,7 @@
 VDRPLUGINCREATOR(cPlugin$PLUGIN_CLASS); // Don't touch this!
 };
 
-$PLUGINDIR = "$PLUGINS_SRC/$PLUGIN_NAME";
+$PLUGINDIR = "$PLUGINS_SRC/$PLUGIN_NAME-$PLUGIN_VERSION";
 
 die "The directory $PLUGINS_SRC doesn't exist!\n" unless (-d "$PLUGINS_SRC");
 die "A plugin named '$PLUGIN_NAME' already exists in $PLUGINS_SRC!\n" if (-e "$PLUGINDIR");
@@ -260,7 +260,8 @@
 * fill in the code skeleton in "$PLUGIN_NAME.c" to implement your plugin function
 * add further source files if necessary
 * adapt the "Makefile" if necessary
-* do "make plugins" from the VDR source directory to build your plugin
+* do "make all DVBDIR=/usr/include VDRDIR=/usr/include/vdr LIBDIR=." from the 
+  source directory to build your plugin
 
 };
 
